src/main/java/it/aboutbits/springboot/toolbox/type/ScaledBigDecimal.java

```diff
@@ -0,0 +1,233 @@
+        return new ScaledBigDecimal(this.value().divide(other.value(), MATH_CONTEXT));
+    }
+
+    @NotNull
+    public ScaledBigDecimal add(@NonNull BigDecimal other) {
+        return new ScaledBigDecimal(this.value().add(other));
+    }
+
+    @NotNull
+    public ScaledBigDecimal subtract(@NonNull BigDecimal other) {
+        return new ScaledBigDecimal(this.value().subtract(other));
+    }
+
+    @NotNull
+    public ScaledBigDecimal multiply(@NonNull BigDecimal other) {
+        return new ScaledBigDecimal(this.value().multiply(other));
+    }
+
+    @NotNull
+    public ScaledBigDecimal divide(@NonNull BigDecimal other) {
+        return new ScaledBigDecimal(this.value().divide(other, MATH_CONTEXT));
+    }
+
+    @NotNull
+    public ScaledBigDecimal remainder(@NonNull ScaledBigDecimal divisor) {
+        return new ScaledBigDecimal(this.value().remainder(divisor.value(), MATH_CONTEXT));
+    }
+
+    @NotNull
+    public ScaledBigDecimal remainder(@NonNull BigDecimal divisor) {
+        return new ScaledBigDecimal(this.value().remainder(divisor, MATH_CONTEXT));
+    }
+
+    @NotNull
+    public ScaledBigDecimal sqrt() {
+        return new ScaledBigDecimal(this.value().sqrt(MATH_CONTEXT));
+    }
+
+    @NotNull
+    public ScaledBigDecimal pow(int n) {
+        return new ScaledBigDecimal(this.value().pow(n, MATH_CONTEXT));
+    }
+
+    @NotNull
+    public ScaledBigDecimal abs() {
+        return new ScaledBigDecimal(this.value().abs(MATH_CONTEXT));
+    }
+
+    @NotNull
+    public ScaledBigDecimal negate() {
+        return new ScaledBigDecimal(this.value().negate(MATH_CONTEXT));
+    }
+
+    @NotNull
+    public ScaledBigDecimal min(@NonNull ScaledBigDecimal val) {
+        return new ScaledBigDecimal(this.value().min(val.value()));
+    }
+
+    @NotNull
+    public ScaledBigDecimal min(@NonNull BigDecimal val) {
+        return new ScaledBigDecimal(this.value().min(val));
+    }
+
+    @NotNull
+    public ScaledBigDecimal max(@NonNull ScaledBigDecimal val) {
+        return new ScaledBigDecimal(this.value().max(val.value()));
+    }
+
+    @NotNull
+    public ScaledBigDecimal max(@NonNull BigDecimal val) {
+        return new ScaledBigDecimal(this.value().max(val));
+    }
+
+    @NotNull
+    public BigDecimal toBigDecimal(int scale) {
+        return this.value().setScale(scale, RoundingMode.HALF_UP);
+    }
+
+    @NotNull
+    public ScaledBigDecimal roundToScale(int scale) {
+        return new ScaledBigDecimal(this.value().setScale(scale, RoundingMode.HALF_UP));
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+
+        if (o instanceof ScaledBigDecimal other) {
+            return this.value().compareTo(other.value()) == 0;
+        }
+
+        return false;
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(MATH_CONTEXT, value());
+    }
```

Comment:
Are you sure about this hashCode implementation?

Would it make sense to use the hashCode method of the wrapped BigDecimal?
```suggestion
    @Override
    public int hashCode() {
        return value().hashCode();
    }
```